solver.rb

```diff
@@ -0,0 +1,27 @@
+class Solver
+  def factorial(num)
+    if num.negative?
+      nil
```

Comment:
- [ ] When a number is negative an exception is expected to be raised. Kindly fix this and the associated test.